repos/system_upgrade/el7toel8/actors/checkpcidrivers/tests/unit_test_checkpcidrivers.py

```diff
@@ -0,0 +1,367 @@
+    )
+    monkeypatch.setattr(
+        reporting,
+        "create_report",
+        create_report_mocked(),
+    )
+    monkeypatch.setattr(
+        CurrentActorMocked,
+        "produce",
+        produce_mocked(),
+    )
+    checkpcidrivers_main()
+    if inhibit_upgrade:
+        assert len(CurrentActorMocked.produce.model_instances) == 1
+        assert CurrentActorMocked.produce.model_instances[0].report[
+            "flags"
+        ] == ["inhibitor"]
+    if num_msgs and not inhibit_upgrade:
+        assert (
+            "flags" not in CurrentActorMocked.produce.model_instances[0].report
+        )
+    assert len(CurrentActorMocked.produce.model_instances) == num_msgs
+
+
+RestrictedPCIDevicePrefilled = partial(
+    RestrictedPCIDevice,
+    pci_id="",
+    driver_name="",
+    device_name="",
+    comment="",
+    available_rhel7=1,
+    available_rhel9=0,
+    supported_rhel7=1,
+    supported_rhel9=0,
+)
+
+
+def get_test_case_for_testing_report(
+    devices, available_rhel8, supported_rhel8
+):
+    return tuple(d for d in devices), {
+        d: RestrictedPCIDevicePrefilled(
+            available_rhel8=available_rhel8,
+            supported_rhel8=supported_rhel8,
+        )
+        for d in devices
+    }
+
+
+@pytest.mark.parametrize(
+    (
+        "restricted_driver_names_on_host",
+        "restricted_devices_drivers",
+        "restricted_pci_ids_on_host",
+        "restricted_devices_pcis",
+        "inhibit_upgrade",
+        "exp_line_length",
+    ),
+    [
+        get_test_case_for_testing_report(("d1", "d2"), 0, 0)
+        + get_test_case_for_testing_report(("p1", "p2"), 0, 0)
+        + (True, 15),
+        get_test_case_for_testing_report(("d1", "d2"), 1, 0)
+        + get_test_case_for_testing_report(("p1", "p2"), 1, 0)
+        + (True, 8),
+        get_test_case_for_testing_report(("d1", "d2"), 0, 1)
+        + get_test_case_for_testing_report(("p1", "p2"), 0, 1)
+        + (True, 8),
+        get_test_case_for_testing_report(("d1", "d2"), 1, 0)
+        + get_test_case_for_testing_report(("p1", "p2"), 0, 1)
+        + (True, 9),
+    ],
+)
+def test_render_report(
+    monkeypatch,
+    restricted_driver_names_on_host,
+    restricted_pci_ids_on_host,
+    restricted_devices_drivers,
+    restricted_devices_pcis,
+    inhibit_upgrade,
+    exp_line_length,
+    caplog,
+):
+    """Check report appearance."""
+    monkeypatch.setattr(
+        reporting,
+        "create_report",
+        create_report_mocked(),
+    )
+    reporting.create_report(
+        render_report(
+            restricted_driver_names_on_host=restricted_driver_names_on_host,
+            restricted_pci_ids_on_host=restricted_pci_ids_on_host,
+            restricted_devices_drivers=restricted_devices_drivers,
+            restricted_devices_pcis=restricted_devices_pcis,
+            inhibit_upgrade=inhibit_upgrade,
+        )
+    )
+    logger.info(reporting.create_report.report_fields["summary"])
```

Comment:
@fernflower Inna, now the reports has the follwoing appearance:
```
-------------------------------- live log call ---------------------------------
| 11:43:36 | unit_test_checkpcidrivers | INFO | unit_test_checkpcidrivers.py | The following drivers are restricted on RHEL 8 system:
	 Unavailable in RHEL8:
		- driver names:
			- d1
			- d2
		- pci ids:
			- p1
			- p2
	 Unsupported in RHEL8:
		- driver names:
			- d1
			- d2
		- pci ids:
			- p1
			- p2
PASSED
-------------------------------- live log call ---------------------------------
| 11:43:36 | unit_test_checkpcidrivers | INFO | unit_test_checkpcidrivers.py | The following drivers are restricted on RHEL 8 system:
	 Unsupported in RHEL8:
		- driver names:
			- d1
			- d2
		- pci ids:
			- p1
			- p2
PASSED
-------------------------------- live log call ---------------------------------
| 11:43:36 | unit_test_checkpcidrivers | INFO | unit_test_checkpcidrivers.py | The following drivers are restricted on RHEL 8 system:
	 Unavailable in RHEL8:
		- driver names:
			- d1
			- d2
		- pci ids:
			- p1
			- p2
PASSED
-------------------------------- live log call ---------------------------------
| 11:43:36 | unit_test_checkpcidrivers | INFO | unit_test_checkpcidrivers.py | The following drivers are restricted on RHEL 8 system:
	 Unavailable in RHEL8:
		- pci ids:
			- p1
			- p2
	 Unsupported in RHEL8:
		- driver names:
			- d1
			- d2
PASSED
```

Reply:
Much clearer to the user, thanks.
Although as a user I wouldn't bother with the details and having one driver\pci_id in both unavailable and unsupported section can be confusing (like it is not obvious how the user can mediate this problem anyway).
If the intention is to give full information, I would rather go with Unavailable & Unsupported, Unavailable, Unsupported grouping.